ethereum/spec/src/property-test/java/tech/pegasys/teku/spec/datastructures/blocks/BeaconBlockBodySupplier.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright ConsenSys Software Inc., 2022
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.spec.datastructures.blocks;
+
+import net.jqwik.api.Arbitraries;
+import net.jqwik.api.Arbitrary;
+import net.jqwik.api.ArbitrarySupplier;
+import net.jqwik.api.Combinators;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.SpecMilestone;
+import tech.pegasys.teku.spec.TestSpecFactory;
+import tech.pegasys.teku.spec.datastructures.blocks.blockbody.BeaconBlockBody;
+import tech.pegasys.teku.spec.networks.Eth2Network;
+import tech.pegasys.teku.spec.util.DataStructureUtil;
+
+public class BeaconBlockBodySupplier implements ArbitrarySupplier<BeaconBlockBody> {
+  @Override
+  public Arbitrary<BeaconBlockBody> get() {
+    Arbitrary<Integer> seed = Arbitraries.integers();
+    Arbitrary<SpecMilestone> milestone = Arbitraries.of(SpecMilestone.class);
+    Arbitrary<Eth2Network> network = Arbitraries.of(Eth2Network.class);
+    Arbitrary<Spec> spec = Combinators.combine(milestone, network).as(TestSpecFactory::create);
+    Arbitrary<DataStructureUtil> dsu = Combinators.combine(seed, spec).as(DataStructureUtil::new);
```

Comment:
Would it be possible and make sense to use a `DataStructureUtilSupplier` to create the `DataStructureUtil` instance and not need to duplicate this in each supplier?

Could use a common base class like:
```java
public abstract class DataStructureSupplier<T> implements ArbitrarySupplier<T> {
  private final Function<DataStructureUtil, T> accessor;

  protected DataStructureSupplier(final Function<DataStructureUtil, T> accessor) {
    this.accessor = accessor;
  }

  @Override
  public Arbitrary<T> get() {
    Arbitrary<Integer> seed = Arbitraries.integers();
    Arbitrary<SpecMilestone> milestone = Arbitraries.of(SpecMilestone.class);
    Arbitrary<Eth2Network> network = Arbitraries.of(Eth2Network.class);
    Arbitrary<Spec> spec = Combinators.combine(milestone, network).as(TestSpecFactory::create);
    Arbitrary<DataStructureUtil> dsu = Combinators.combine(seed, spec).as(DataStructureUtil::new);
    return dsu.map(accessor);
  }
}
```
 and then the supplier would be like:
 ```java
public class BeaconBlockHeaderSupplier extends DataStructureSupplier<BeaconBlockHeader> {
  public BeaconBlockHeaderSupplier() {
    super(DataStructureUtil::randomBeaconBlockHeader);
  }
}
```

Reply:
Ah - the filtering on milestone changes in a few cases so might need to be able to pass the minimum required milestone up to the `DataStructureSupplier` class too.